python-package/lightgbm/basic.py

```diff
@@ -361,7 +361,7 @@ def _get_all_param_aliases() -> Dict[str, Set[str]]:
                 ptr_string_buffer))
         aliases = json.loads(
             string_buffer.value.decode('utf-8'),
-            object_hook=lambda obj: {k: set(v) | {k} for k, v in obj.items()}
+            object_hook=lambda obj: {k: [k] + v for k, v in obj.items()}
```

Comment:
I understand why this PR is proposing switching this value to a list (since sets don't preserve insert order). I'm ok with that, given that the number of items in any one of these lists is so small.

But can you please update this method's return type hint? I think it is now `Dict[str, List[str]]`.